ChapmanUniversity1.0/Controllers/SemestersController.cs

```diff
@@ -53,27 +52,45 @@ public IActionResult Create(Semester semester)
             List<string> seasonsList = new List<string>(Enum.GetNames(typeof(Seasons)));
             ViewData["Seasons"] = new SelectList(seasonsList);
 
-            var course = _unitOfWork.Courses.GetById(semester.CourseId);
-            var semesterExists = _unitOfWork.Semesters.SemesterExists(course.Id, semester.CourseSeason);
+            var semesters = _unitOfWork.SemesterRepository.Get();
 
-            if (!semesterExists)
+            var enumerable = semesters as Semester[] ?? semesters.ToArray();
```

Comment:
You still want to call this variable `semesters` or `semestersList`. Naming it what type it is could be confusing for future uses.